psydac/feec/multipatch/examples/h1_source_pbms_conga_2d.py

```diff
@@ -118,52 +120,57 @@ def solve_h1_source_pbm(
     H0 = HodgeOperator(V0h, domain_h, backend_language=backend_language)
     H1 = HodgeOperator(V1h, domain_h, backend_language=backend_language)
 
-    dH0_m = H0.get_dual_Hodge_sparse_matrix()  # = mass matrix of V0
-    H0_m  = H0.to_sparse_matrix()              # = inverse mass matrix of V0
-    dH1_m = H1.get_dual_Hodge_sparse_matrix()  # = mass matrix of V1
-    # H1_m  = H1.to_sparse_matrix()              # = inverse mass matrix of V1
+    H0_m = H0.to_sparse_matrix()              # = mass matrix of V0
+    dH0_m = H0.get_dual_Hodge_sparse_matrix()  # = inverse mass matrix of V0
+    H1_m = H1.to_sparse_matrix()              # = mass matrix of V1
+    dH1_m = H1.get_dual_Hodge_sparse_matrix()  # = inverse mass matrix of V1
 
     print('conforming projection operators...')
-    # conforming Projections (should take into account the boundary conditions of the continuous deRham sequence)
-    cP0 = derham_h.conforming_projection(space='V0', hom_bc=True, backend_language=backend_language)
-    cP0_m = cP0.to_sparse_matrix()
-    # cP1 = derham_h.conforming_projection(space='V1', hom_bc=True, backend_language=backend_language)
-    # cP1_m = cP1.to_sparse_matrix()
+    # conforming Projections (should take into account the boundary conditions
+    # of the continuous deRham sequence)
+    cP0_m = construct_h1_conforming_projection(V0h, hom_bc=True)
+    # cP1_m = construct_hcurl_conforming_projection(V1h, hom_bc=True)
 
     if not os.path.exists(plot_dir):
         os.makedirs(plot_dir)
 
     def lift_u_bc(u_bc):
         if u_bc is not None:
-            print('lifting the boundary condition in V0h...  [warning: Not Tested Yet!]')
-            # note: for simplicity we apply the full P1 on u_bc, but we only need to set the boundary dofs
+            print(
+                'lifting the boundary condition in V0h...  [warning: Not Tested Yet!]')
+            # note: for simplicity we apply the full P1 on u_bc, but we only
+            # need to set the boundary dofs
             u_bc = lambdify(domain.coordinates, u_bc)
-            u_bc_log = [pull_2d_h1(u_bc, m) for m in mappings_list]
-            # it's a bit weird to apply P1 on the list of (pulled back) logical fields -- why not just apply it on u_bc ?
+            u_bc_log = [pull_2d_h1(u_bc, m.get_callable_mapping())
+                        for m in mappings_list]
+            # it's a bit weird to apply P1 on the list of (pulled back) logical
+            # fields -- why not just apply it on u_bc ?
```

Comment:
this comment should be turned into a todo: improve the `projectors` interface in multipatch deRham sequences so that it applies to physical functions

Reply:
See https://github.com/pyccel/psydac/pull/320#discussion_r1531796164.